php-checks/src/test/resources/checks/TooManyFunctionParametersCheck.php

```diff
@@ -36,3 +36,55 @@ function f($p1, $p2, $p3, $p4, $p5, $p6, $p7, $p8) {
   function g($p1, $p2, $p3, $p4, $p5, $p6, $p7, $p8) { // Noncompliant
   }
 }
+
+class OnePromotedProperty {
+  function __construct(
+    public $p1,
+    $p2) {
+  }
+}
+
+class OnlyPromotedProperties {
+  function __construct(
+    public $p1,
+    PUBLIC $p2,
+    protected $p3,
+    PROTECTED $p4,
+    private $p5,
+    PRIVATE $p6,
+    readonly PRIVATE $p7,
+    PRIVATE READONLY $p8) {
+  }
+}
```

Comment:
I would suggest a test where there are x non-promoted properties where x+1 is the threshold for raising.
Then I would add a promoted property for every flavor there is (public, protected and private)
Basically if any of these would not register as promoted property there would be an issue raised.

The test is not essential as I think it's tested already good, but would be nice to have IMO